quantecon/game_theory/polymatrix_game.py

```diff
@@ -0,0 +1,243 @@
+    and player number `b` is the column player.
+
+    Attributes
+    ----------
+    number_of_players : int
+        Number of players.
+
+    nums_actions : Sequence[int]
+        The number of actions available to each player.
+
+    polymatrix : Mapping[ tuple[int, int], Matrix ]
+        Maps each pair of player numbers to a matrix.
+
+    """
+
+    def __str__(self) -> str:
+        str_builder = "Polymatrix Game:\n"
+        for k, v in self.polymatrix.items():
+            str_builder += str(k) + ":\n"
+            str_builder += str(v) + "\n\n"
+        return str_builder.rstrip()
+
+    def __init__(
+            self,
+            number_of_players: int,
+            nums_actions: Sequence[int],
+            polymatrix: Mapping[
+                tuple[int, int],
+                Matrix
+            ]
+    ) -> None:
+        self.N = number_of_players
+        self.nums_actions = nums_actions
+        self.polymatrix = polymatrix
+
+    @classmethod
+    def from_nf(
+        cls,
+        nf: NormalFormGame,
+        is_polymatrix: bool = True
+    ) -> Self:
+        """
+        Creates a Polymatrix from a Normal Form Game.
+
+        Precise if possible; many Normal Form Games are not representable
+        precisely with a Polymatrix.
+        With payoffs (not costs).
+
+        Parameters
+        ----------
+        nf : NormalFormGame
+            Normal Form Game to convert.
+
+        is_polymatrix : bool, optional
+            Is the Normal Form Game precisely convertible to a
+            Polymatrix Game. By default True
+
+        Returns
+        -------
+        Self
+            The Polymatrix Game.
+        """
+        polymatrix_builder = {
+            (p1, p2): np.full(
+                (nf.nums_actions[p1], nf.nums_actions[p2]), -np.inf)
+            for p1 in range(nf.N)
+            for p2 in range(nf.N)
+            if p1 != p2
+        }
+        for p1 in range(nf.N):
+            for a1 in range(nf.nums_actions[p1]):
+                payoffs = hh_payoff_player(
+                    nf, p1, a1, is_polymatrix=is_polymatrix)
+                for ((p2, a2), payoff) in payoffs.items():
+                    polymatrix_builder[(p1, p2)][a1][a2] = payoff
+
+        return cls(nf.N, nf.nums_actions, polymatrix_builder)
+
+    def get_player(self, player_idx: int) -> Player:
+        """
+        Calculates the payoff function of a player.
+
+        Parameters
+        ----------
+        player_idx : int
+            Player number who we want to extract.
+
+        Returns
+        -------
+        Player
+            Player object which has the player's payoff function.
+        """
+        self.N
+        opps = tuple(range(player_idx+1, self.N)) + tuple(range(player_idx))
+        newaxes = np.full((self.N-1, self.N-1), np.newaxis)
+        np.fill_diagonal(newaxes, slice(None))
+        payoff_array = sum([
+            self.polymatrix[(player_idx, opps[j])][:, *newaxes[j]]
```

Comment:
It turned out that this expression `[:, *newaxes[j]]` is not supported in Python 3.10:

> ```
> E     File "/home/runner/work/QuantEcon.py/QuantEcon.py/quantecon/game_theory/polymatrix_game.py", line 213
> E       self.polymatrix[(player_idx, opps[j])][:, *tuple(newaxes[j])]
> E                                                 ^
> E   SyntaxError: invalid syntax
> ```

(https://github.com/QuantEcon/QuantEcon.py/actions/runs/11947385419/job/33303244882#step:7:70)

Please apply the diff 26cf5a6